specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/stable/2020-03-01/machineLearningServices.json

```diff
@@ -1228,6 +1228,283 @@
           }
         }
       }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/Microsoft.MachineLearningServices/workspaces/{workspaceName}/linkedWorkspaces": {
+      "get": {
+        "tags": [
+          "LinkedWorkspaces"
+        ],
+        "x-ms-examples": {
+          "ListLinkedWorkspaces": {
+            "$ref": "./examples/listLinkedWorkspaces.json"
+          }
+        },
+        "operationId": "LinkedWorkspaces_List",
+        "description": "List linked workspaces under a workspace per some query criteria.",
+        "parameters": [
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "#/parameters/WorkspaceNameParameter"
+          },
+          {
+            "name": "link-subscriptionId",
+            "in": "query",
+            "type": "string",
+            "description": "The subscriptionid of the linked workspace."
+          },
+          {
+            "name": "link-resource-group",
+            "in": "query",
+            "type": "string",
+            "description": "The resource group name of the linked workspace."
+          },
+          {
+            "name": "link-resource-name",
+            "in": "query",
+            "type": "string",
+            "description": "The resource name of the linked workspace."
+          },
+          {
+            "name": "link-resourceId",
+            "in": "query",
+            "type": "string",
+            "description": "The resourceId of the linked workspace."
+          },
+          {
+            "name": "link-type",
+            "in": "query",
+            "type": "string",
+            "description": "The type of the linked workspace.",
+            "enum": [
+              "Synapse"
+            ],
+            "x-ms-enum": {
+              "name": "LinkType",
+              "modelAsString": true
+            }
+          },
+          {
+            "name": "uai-resourceId",
+            "in": "query",
+            "type": "string",
+            "description": "The resourceId of the user assigned identity for the linked workspace."
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Successfully retrieved the linked workspaces.",
+            "schema": {
+              "type": "array",
+              "description": "array of linked workspaces.",
+              "items": {
+                "$ref": "#/definitions/LinkedWorkspace"
```

Comment:
>LinkedWorkspace [](start = 39, length = 15)

This is a list call, but it appears to be returning a single workspace.

Reply:
it's an array, the items of the array is #/definitions/LinkedWorkspace

Reply:
Yes, you're right I missed the `array` part here. However, that is not a valid list response. ARM uses the OData list format, which looks like the following in JSON:
```
{
  "value": [
    { object 1 },
    { object 2 },
    ....
  ],
  "nextLink": "<link to the next page or null>"
}
```

This is documented at [https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/resource-api-reference.md#get-resource](https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/resource-api-reference.md#get-resource) under the "Paging Response Body" section.

---
In reply to: [402721411](https://github.com/Azure/azure-rest-api-specs/pull/8918#discussion_r402721411) [](ancestors = 402721411)